core/src/bitmap/bitmap_data.rs

```diff
@@ -69,11 +69,10 @@ impl Color {
 
         let old_alpha = if transparency { self.alpha() } else { 255 };
 
-        let a = old_alpha as f64 / 255.0;
-
-        let r = (self.red() as f64 * a).round() as u8;
-        let g = (self.green() as f64 * a).round() as u8;
-        let b = (self.blue() as f64 * a).round() as u8;
+        let a = old_alpha as u32;
```

Comment:
Does this make sense?
```rs
let old_alpha = if transparency { self.alpha() } else { 255 };
let premultiply = |x| {
    let x = x as u16 * old_alpha as u16 + 127;
    ((x + (x / 255) + 1) / 255) as u8
};

let a = premultiply(self.alpha());
let r = premultiply(self.red());
let g = premultiply(self.green());
let b = premultiply(self.blue());
Self::argb(a, r, g, b)
```
This may fix the accuracy issues described in the comment.

Reply:
Otherwise, do you think this is still blocking the merge?

Reply:
Let's merge this as-is for now, since a deeper investigation is needed.